lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -1228,164 +1237,201 @@ namespace Js
         }
-        return 0;
+
+        return flags;
     }
+#endif
 
-    // Takes arguments as follows(all required):
-    //     - [1] - String 1 for comparison
-    //     - [2] - String 2 for comparison
-    //     - [3] - Locale string (or undefined)
-    //     - [4] - Sensitivity string (or undefined)
-    //     - [5] - IgnorePunctuation boolean (or undefined)
-    //     - [6] - Numeric boolean (or undefined)
-    Var IntlEngineInterfaceExtensionObject::EntryIntl_CompareString(RecyclableObject* function, CallInfo callInfo, ...)
+    Var IntlEngineInterfaceExtensionObject::EntryIntl_CollatorGetCollation(RecyclableObject* function, CallInfo callInfo, ...)
     {
         EngineInterfaceObject_CommonFunctionProlog(function, callInfo);
 
-        if (args.Info.Count < 7 || !JavascriptString::Is(args.Values[1]) || !JavascriptString::Is(args.Values[2]))
+#if defined(INTL_ICU)
+        if (args.Info.Count < 2 || !JavascriptString::Is(args.Values[1]))
         {
-            // CompareStringEx of undefined or non-strings is undefined
             return scriptContext->GetLibrary()->GetUndefined();
         }
 
-        DWORD compareFlags = 0;
-        JavascriptString* str1 = JavascriptString::FromVar(args.Values[1]);
-        JavascriptString* str2 = JavascriptString::FromVar(args.Values[2]);
+        JavascriptString *locale = JavascriptString::FromVar(args.Values[1]);
+        utf8::WideToNarrow locale8(locale->GetSz(), locale->GetLength());
 
-        WCHAR defaultLocale[LOCALE_NAME_MAX_LENGTH];
-        const char16 *givenLocale = nullptr;
-        defaultLocale[0] = '\0';
+        char collation8[LOCALE_NAME_MAX_LENGTH] = { 0 };
+        AssertMsg(CollatorGetCollation(locale8, collation8, _countof(collation8)), "Could not get resolved collation");
+
+        utf8::NarrowToWide collation(collation8);
 
-        if (!JavascriptOperators::IsUndefinedObject(args.Values[3]))
+        return JavascriptString::NewCopySz(collation, scriptContext);
+#else
+        return scriptContext->GetLibrary()->GetNull();
+#endif
+    }
+
+    Var IntlEngineInterfaceExtensionObject::EntryIntl_CompareString(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        EngineInterfaceObject_CommonFunctionProlog(function, callInfo);
+
+        if (args.Info.Count < 3 || !JavascriptString::Is(args.Values[1]) || !JavascriptString::Is(args.Values[2]))
         {
-            if (!JavascriptString::Is(args.Values[3]))
-            {
-                return scriptContext->GetLibrary()->GetUndefined();
-            }
-            givenLocale = JavascriptString::FromVar(args.Values[3])->GetSz();
+            JavascriptError::MapAndThrowError(scriptContext, E_INVALIDARG);
         }
 
-        if (!JavascriptOperators::IsUndefinedObject(args.Values[4]))
+        JavascriptString *str1 = JavascriptString::FromVar(args.Values[1]);
+        JavascriptString *str2 = JavascriptString::FromVar(args.Values[2]);
+        const char16 *locale = nullptr; // args[3]
```

Comment:
>// args[3] [](start = 40, length = 10)

Add a blank line above this line since these 5 locals are different semantically.

Are any of these 5 values used only inside of if (Count > 3) ?

Reply:
Nope, theyre all relevant regardless (the default values I give are important)